auction_keeper/main.py

```diff
@@ -397,11 +397,11 @@ def check_flop(self):
                         if self.vow.woe() - joy >= sump:
                             break
 
-            # use heal() for reconciling the remaining joy
+            # Kiss again to reconcile remaining joy
             joy = self.vat.dai(self.vow.address)
             if Rad(0) < joy <= self.vow.woe():
-                self.vow.heal(joy).transact(gas_price=self.gas_price)
-                # heal() changes joy and woe (the balance of surplus and debt)
+                # kiss() again to reduce ash and implicitly heal, resetting joy
```

Comment:
I don't think this is quite right, although I haven't had time to think through it holistically.

`kiss` only works if `Ash` actually exists.

So depending on the relative sizes of `joy`, `woe`, and `Ash, you might need
1) A call to `heal`
2) A call to `kiss`
3) A call to *both*

Maybe the preconditions to this already ensure there will be `Ash`, but from a cursory glance I don't think so.